whereby/browser-sdk
# Handle recording and streaming states

* Fix the bug where the recorder client appears in the list of participant for the client who joins the recorder meeting
* Add cloud recording and streaming states to the room connection state

### Test plan

_Starting/stopping cloud recording and streaming is not available in SDK yet. These can be controlled by the host in the regular pwa room._

1. Join a room with an sdk client where cloud recording is in progress => the sdk client shouldn't see a "phantom" guest (with `recorder` role)
2. Join a room with an sdk client where streaming is in progress => the sdk client shouldn't see a `streamer` guest 
3. Start/stop recording by the host when the sdk client is in the room => check `roomConnection.state.cloudRecording` => it should show `startedAt` and `status` value changes
4. Start/stop streaming by the host when the sdk client is in the room => check `roomConnection.state.streaming` => it should show `startedAt` and `status` value changes

### Screenshot

<img width="1144" alt="Screenshot 2023-08-31 at 13 29 07" src="https://github.com/whereby/browser-sdk/assets/1159931/fa68609f-3c2f-4de3-a465-6078ddeb75bf">
